op-bindings/bindgen/remote_handlers_test.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestRemoveDeploymentSalt(t *testing.T) {
+	generator := bindGenGeneratorRemote{}
+
+	for _, tt := range removeDeploymentSaltTests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, _ := generator.removeDeploymentSalt(tt.deploymentData, tt.deploymentSalt)
+			if diff := cmp.Diff(tt.expected, got); diff != "" {
+				t.Errorf("%s mismatch (-want +got):\n%s", tt.name, diff)
+			}
+		})
+	}
+}
+
+func TestRemoveDeploymentSaltFailures(t *testing.T) {
+	generator := bindGenGeneratorRemote{}
+
+	for _, tt := range removeDeploymentSaltTestsFailures {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := generator.removeDeploymentSalt(tt.deploymentData, tt.deploymentSalt)
+
+			if err == nil {
+				t.Errorf("Expected error: %s but didn't receive it", tt.expectedError)
+				return
```

Comment:
We commonly use the `stretchr/testify` testing library for assertions. If the errors should have equal strings you could also do `require.Equal(t, err.String(), t.expectedError.String())`
```suggestion
		t.Run(tt.name, func(t *testing.T) {
			_, err := generator.removeDeploymentSalt(tt.deploymentData, tt.deploymentSalt)
                         require.Error(t, err, "expected %s but didn't receive it", tt.expectedError)
```